UnityProject/Assets/Scripts/Lighting/LightSource.cs

```diff
@@ -89,16 +65,22 @@ private void EnsureInit()
 			switchState = InitialState == LightMountState.On;
 	}
 
-	public override void OnStartServer()
+	private void OnDirectionChange(Orientation newDir)
 	{
-		EnsureInit();
-		base.OnStartServer();
-		mState = InitialState;
+		SetSprites();
+	}
+
+	public void OnSpawnServer(SpawnInfo info)
+	{
+		if (!info.SpawnItems)
+		{
+			mState = LightMountState.MissingBulb;
+		}
 	}
 
 	public override void OnStartClient()
 	{
-		EnsureInit();
+		//EnsureInit();
```

Comment:
Did you test it with late joining client? This int method there to make sure late joiners will have the same state of lights.

Reply:
yeah ensureinit() was being spamcalled, I just forgot to remove this one after commenting it out